### PR TITLE
Enable circleCI machine exec

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Enable machine executor on CircleCI

CircleCI machine specs
```bash
Linux default-8e89cf48-a14c-4751-8bd5-2f739397ef81 4.4.0-96-generic #119~14.04.1-Ubuntu SMP Wed Sep 13 08:40:48 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
NAME="Ubuntu"
VERSION="14.04.5 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.5 LTS"
VERSION_ID="14.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
nproc = 2
go version = 1.9
```

This is a first PR in a series to enhance singularity test suite

Signed-off-by: ArangoGutierrez <eduardo@sylabs.io>

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by jmstover — APPROVED** [review]
This looks okay to me. With the CI stuff.... I'm not the one to ask. ;)

**Comment by jscook2345** [comment]
This doesn't seem like it should work without the installation of `dep` in each of the workflows. How is it that you don't need to install `dep` until the `singularity_test` workflow?

**Comment by jscook2345** [comment]
@ArangoGutierrez noted that `dep` is in the golang base image used in get_source. So it really only needs to be installed when we go to the machine executor.

**Review by jscook2345 — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,100 +24,117 @@ jobs:
...
-            fi
-      - save_cache:
-          key: dep-cache-{{ checksum "Gopkg.lock" }}
-          paths:
-            - vendor
-      - run:
-          name: Install C dependencies
-          command: apk add --update alpine-sdk automake libtool linux-headers linux-vanilla libarchive-dev util-linux-dev libuuid openssl-dev gawk
+            mkdir -p ${GOBIN}
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (kzap) [inline .circleci/config.yml:109]: why do we install dep even though its already fetched from the `get_source` pipeline?
wouldnt that mean we already have our `/vendor` directory populated before this and thus no longer need to install dep via `curl` ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,100 +24,117 @@ jobs:
...
+          name: Build Singularity
           command: |-
-            apk add --update git
+            ./mconfig -p /usr/local
+            make -j$(nproc) -C ./builddir all
+            go get -u golang.org/x/lint/golint
+            make -j$(nproc) -C ./builddir check
       - run:
-          name: checkout
+          name: check code
```
Comment (kzap) [inline .circleci/config.yml]: personal linter says capitalize or dont capitalize :) but be consistent :)

**Review by kzap — COMMENTED** [review]
question about this line

File: .circleci/config.yml (hunk 1 above)
Comment (kzap) [reply .circleci/config.yml:109]: From Slack:
> Andre: yes but why do we need to install it on the machine? is it needed for running go build?
i thought its not needed as long as `/vendor` has the files already?

> Eduardo Arango [3:09 AM]
`./mconfig` will fail if it is not in path
is just to make `./mconfig` happy
when we merge Cedric PR on the vendor folder
I can then modify that part

**Review by kzap — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (ArangoGutierrez) [reply .circleci/config.yml:109]: this will addressed once #2200 is merged

File: .circleci/config.yml (hunk 2 above)
Comment (ArangoGutierrez) [reply .circleci/config.yml]: will fix

**Pull request merged**